src/components/example-viewer/example-viewer.ts

```diff
@@ -0,0 +1,24 @@
+import {Component, Input} from '@angular/core';
+import {Portal, ComponentPortal} from '@angular/material';
+import {EXAMPLE_COMPONENTS} from './example-module';
+
+
+@Component({
+  selector: 'example-viewer',
+  template: `{{example}}<template [portalHost]="selectedPortal"></template>`,
```

Comment:
It's portals all the way down
![](https://2.bp.blogspot.com/-R5MLwwBC2GU/VzAWOJMT2LI/AAAAAAAANCU/aQ0EPBXByckj9W86sf-f_1-qk7myzBPXQCKgB/s1600/Infinite_Portals.gif)